### DISCUSSION TITLE
Mehr als 10 Hoymiles... was nun?

### DISCUSSION THREAD

**Discussion by bynight66** [tbnobody/OpenDTU#1953] [Q&A]
Gibt es hier eine Lösung die 99 Wechselrichter unterstützt? Die DTU Pro kann es doch auch... 
Ich habe hier 4 opene DTUs, aber ich schaffe es nicht mal 2 open DTUs gleichzeitig zu betreiben, geschweige denn 4.
Es läuft immer nur eine und die anderen gehen nach kurzer Zeit Offline. Wenn ich davon eine resette geht dafür dann kurz darauf die vorher laufende offline.
Hat jemand eine Idee woran das liegt? Hat überhaupt jemand 2 Open DTUs zum laufen bekommen? Sieht irgendwie aus wie eine Sackgasse...

Gruß,
Ron

**Comment by MetaChuh** [comment]
@bynight66 

> Sieht irgendwie aus wie eine Sackgasse...

kommt darauf an, ob ein sack in der gasse wohnt 😉

wieviele inverter benötigst du pro opendtu ?
bis 14 kannst du dir freischalten, wenn du sebst compilierst, und vorher in der datei: `OpenDTU/include/Configuration.h`
`#define INV_MAX_COUNT 10`
gegen:
`#define INV_MAX_COUNT 14`
ersetzt.
danach ist schluss, wegen der config größe.

greetings,
metachuh

**Reply by bynight66** [reply]
Hallo MetaChuh,

aktuell hab ich 31, später werden es mindestens 40.
Danke für den Tip mit der 14. Aber selber compilieren ist jetzt nicht so meines. Sollte die Begrenzung wirklich an einer lapidaren config Größe liegen?

Danke und Gruß,
Ronald

**Comment by ghost** [comment]


**Reply by bynight66** [reply]
Hi Schneeer,

der Tip war gut! Haben alle dieselbe Seriennummer. Weiß nach dem updaten nicht mehr ob es daher rührt weil ich immer dieselbe config eingespielt habe oder ob es daher rührt weil der Hersteller immer dasselbe Image aufgespielt hat. 
Ich denke allerdings letzteres weil ich zu Beginn meine erstellte Copy Config mit der config einer neuen unkopierten DTU verglichen habe um eben zu vermeiden daß ich irgendwelche DTU spezifischen Daten wie eine einmalige Seriennummer überschreibe. Die Configs waren aber identisch, also keine unterschiedlichen Seriennummern.

Du hast 2 DTUs an einem Wechselrichter? Irgendwo schreibt der Entwickler daß man das nicht machen soll, weil die Datenströme durcheinander kommen. Zu Beginn hatte ich dieselben 4 Wechselrichter auf allen 4 DTUs. Um das als Ursache auszuschließen habe ich allen DTUs unterschiedliche Wechselrichter gegeben. In dem Zusammenhang habe ich mir meine DTU Pro näher angesehen und festgestellt daß ich da plötzlich ein Haufen WR hatte die sporadisch 0W zeigten. Das gibt es sonst nicht. Bedauerlicherweise funktioniert Parallelbetrieb so also  überhaupt nicht, weshalb ich die Pro abgeschaltet habe. 

Hab die Seriennummer jetzt überall um eines erhöht, also von 1 bis 5. Da bin ich ja gespannt ob das morgen was ändert... ;-)

Danke und Gruß,
Ronald

**Reply by bynight66** [reply]
Das Wechselrichter Limit ist eigentlich der Hauptgrund weshalb ich mit mit den open DTUs rumärgere... So wie es aktuell aussieht kann ich mir kaum vorstellen daß die DTUs eine Redzuierung zuverlässig an alle 30-40 Wechselrichter übermitteln... 
Die DTU Pro scheint da zuverlässiger zu arbeiten... und gestern habe ich da auch ne interessante Integration gefunden... https://github.com/banny310/hoymiles-dtu-homeassistant-addon?tab=readme-ov-file

**Reply by bynight66** [reply]
Danke Scheeer!
Der Tip mit der Seriennummer war Gold wert! :-))) Aktuell alle 31 Wechselrichter an 5 openDTUs online! Ohne Neustart! Nicht mal ein einzelner offline, was ich die Tage zuvor auch oft hatte...

**Comment by huste511** [comment]
Was genau ist den das Limit/Grund, dass es Open DTU nicht kann?
Braucht es einen bessen ESP, der mit unter der Haube hat?

Mittelfristig hätte ich hier auch ein paar viele Hoymiles ... und eine ESP-Farm wäre nicht so schön, wenn es anders geht.

**Reply by MetaChuh** [reply]
das opendtu fusion board könnte mehr.
bei ahoydtu wurde für diese environment plattform die inverteranzahl auf 32 erhöht.
bei zeiten teste ich das mal für opendtu aus.

**Reply by bynight66** [reply]
Das frag ich mich auch! 
Meine DTU Pro für 200 Eus ist ein Fliegengewicht, zumindest nicht schwerer als die openDTU mit Gehäuse, und bringt mir bis zu 99 Hoymiles auf einen Bildschirm. 31 sind schon online, 40 werden es mindestens. Das Ganze auch noch in einer sehr angenehmen Ansicht, wo ich viertelstündlich 31 x 2, also 64 Module SEPERAT auf einen Blick sehe und das über den ganzen Tag/ Woche/Monate durchscrollen kann! 

Die Alternative sind aktuell 5 openDTUs für reichlich 300 Euro, für die es nirgends eine übersichtliche Ansicht gibt.  Also der Preis ist schon mal definitiv KEIN Kriterium! Stattdessen ZIG unübersichtliche Screens. Also einmal schalte ich zwischen den DTUs hin und her und dann evtl. in den DTUs noch zwischen den 10 Hoymiles. Zudem büße ich bei Nutzung der DTUs komplett meine übersichtliche DTU Pro ein, weshalb die 5 DTUs auch aus sind. 

Der größte Vorteil ist die hohe Aktualisierungsrate von 5 sec gegenüber 15 min. Aber das liefert mir auch der Shelly 3EM, den ich hinter den Invertern hängen habe und der mit Victron und mit HA kommuniziert. DAS ist also auch nicht das Kriterium für open DTU statt DTU Pro.   
Vllt ist das auch der Grund weshalb die DTU Pro 99 Inverter kann, da sie die ja weniger oft abfragen muß. Aber 1/10 von alle 5 Sec wären immer noch eine Abfrage alle 50sec für 100 HMs an einem ESP... Was am Ende auch reicht. Also warum nur 10 (bzw. 16 bei Ahoi) statt 100??

Der zweite, eigentlich entscheidende Punkt, weshalb ich überhaupt mit den openDTUs angefangen habe, ist die Notwendigkeit/Möglichkeit die Leistung der Inverter runter zu regeln (Victron AC Out Betrieb, Nulleinspeisung...). 
Ansonsten sehe ich in dem open DTU Projekt für größere Anlagen überhaupt keinen Sinn. Da die Nachteile hier die Vorteile weit überwiegen, werde ich die open DTUu zurückschicken und suche nun Projekte die die DTU Pro auslesen. Ich fändes auch weit sinnvoller wenn die Energien die hier in open DTU gesteckt werden, gleichermaßen auch in das Auslesen der DTU Pro per Modbus gesteckt werden würden. Eines habe ich bereits am Laufen und die DTU Pro mit ihrer übersichtlichen Ansicht bleibt dabei voll einsatzfähig, die Pollrate liegt wohl auch bei 2 Minuten oder weniger. Ich hab grad 5 min eingestellt. Das einzige was da noch fehlt ist Regelung der Leistung und die Einbindung in den Cerbo... Hab den Entwickler deshalb schon angesprochen...

**Reply by tbnobody** [reply]
> Meine DTU Pro für 200 Eus ist ein Fliegengewicht, zumindest nicht schwerer als die openDTU mit Gehäuse

Und das Gewicht hat was genau mit den CPU und RAM Specs zu tun? Ein ENIAC von 1946 wog 27 Tonnen. Dein PC zuhause ist vmtl. viel langsamer. Ist er nun besser oder schlechter? 

> , und bringt mir bis zu 99 Hoymiles auf einen Bildschirm. 31 sind schon online, 40 werden es mindestens. Das Ganze auch noch in einer sehr angenehmen Ansicht, wo ich viertelstündlich 31 x 2, also 64 Module SEPERAT auf einen Blick sehe und das über den ganzen Tag/ Woche/Monate durchscrollen kann!

Es steht dir ja frei eine DTU Pro mit Cloud Anbindung zu verwenden. Ist aber ggf. nicht jemeschens Sache.

> Die Alternative sind aktuell 5 openDTUs für reichlich 300 Euro, für die es nirgends eine übersichtliche Ansicht gibt. 

Die Hardware für eine DTU ist für <10€ zu bekommen. Wenn du natürlich etwas fertiges kaufst und nicht basteln möchtest/kannst ist das ebenfalls deine persönliche Entscheidung. Es wird ja niemand gezwungen.

>  Ich fändes auch weit sinnvoller wenn die Energien die hier in open DTU gesteckt werden, gleichermaßen auch in das Auslesen der DTU Pro per Modbus gesteckt werden würden.

Du willst mir jetzt aber nicht vorschreiben was ich in meiner Freizeit mache oder? Das fände ich nämlich schon frech.

**Reply by ghost** [reply]
Was soll das Gemeckere?

Wenn man sich die Mühe macht, und alles was es zur openDTU gibt liest, erfährt man, das der Ursprung war, die Hoymiles Cloud zu umgehen und die Wechselrichter direkt auszulesen. Ziel war anfangs nie 200 Wechselrichter gleichzeitig auszulesen.
Es zwingt doch keiner einen eine open DTU zu verwenden.
Du kannst doch auch eigene Hardware und Software entwickeln und der Öffentlichkeit zur Verfügung stellen.

Und die Entwichklung bleibt sicher nicht bei openDTU für ESP32 stehen. (Außer Meckerer verderben den Entwicklern den Spaß).

Und wenn du mehr wie 300€ für 5 openDTUs  ausgibst bist du doch selber Schuld. Wenn du auf Displays verzichtest, jeweils nur ein Funkmodul nimmst und alles selbst in ein Gehäuse baust, kommst du  locker mit der Hälfte hin. Oder nimm doch die herstellereigene DTU Pro.

Ich bin dankbar, das es die OpenDTU Software hier gibt.
Ich beschäftige mich mit der Software und versuche sie für mich abzuändern, vielleicht kommt auch etwas heraus, was andere nutzen können.
Danke an die freiwilligen selbstlosen Entwickler.

**Reply by Sn0w3y** [reply]
Ich glaube da hat jemand den Begriff "openSource" nicht so ganz verstanden :D

@bynight66 du kannst ja gerne auch einen PR mit den ganzen änderungen erstellen :)

Grüße !

**Reply by huste511** [reply]
@MetaChuh
Lustig, genau das Teil inkl. Gehäuse hatte ich mir letztes Jahr in einem schwachen Moment bestellt.
Liegt bis heute ohne Verwendung rum, da das Gehäuse so klein ist, dass man kein großes Display einbauen kann. ;-)
Das ist doch aber nur die Hardware. Software ist doch OpenDTU bzw. AhoyDTU. Gibt es alternative Firmware die mehr kann?

**Reply by bynight66** [reply]
> Und das Gewicht hat was genau mit den CPU und RAM Specs zu tun? Ein ENIAC von 1946 wog 27 Tonnen. Dein PC zuhause ist vmtl. viel langsamer. Ist er nun besser oder schlechter?

Ich gehe davon aus daß eine CPU die nicht größer ist, nicht mehr Kühlleistung braucht und auch kein größeres Netzteil braucht, auch nicht gravierend mehr Leistung bringt. Also bleibt die ursprüngliche Frage...

> Es steht dir ja frei eine DTU Pro mit Cloud Anbindung zu verwenden. Ist aber ggf. nicht jemeschens Sache.

Ok, den Punkt hatte ich vergessen. Das war wohl auch der Ursprung von AhojDTU bzw. des openDTU Projekts.
Der Punkt ist natürlich nicht von der Hand zu weisen und für Cloudgegner wohl alternativlos.

> Die Hardware für eine DTU ist für <10€ zu bekommen. Wenn du natürlich etwas fertiges kaufst und nicht basteln möchtest/kannst ist das ebenfalls deine persönliche Entscheidung. Es wird ja niemand gezwungen.

Ich hab die Hardware für unter 10 Euro, sogar mehrfach... Da ich aber keine Lust hatte mir das Gebaumel für meinen Versuchsaufbau 5 mal auf den Boden zu hängen und dann nicht zu sehen was passiert, hab ich mir am Ende lieber die fertigen Module geholt, was auch gut war, um zu sehen wie schnell die Module offline gingen... Wenn man mehr als 40 Wechselrichter ansprechen will, und keine Lust auf offene wilde Kabelverdrahtungen hat, wird man mit den Eigenbauteilen kaum noch billiger kommen als mit einer DTU Pro. Wie schon geschrieben, der Preis ist für mich bei der Wahl kein Kriterium.

Und nein, natürlich will ich Dir nicht vorschreiben, was Du in Deiner Freizeit machst. Dein Projekt in allen Ehren! Ist ja für viele Leute eine tolle Sache. Sorry! War mehr so ein allgemeiner Aufruf vllt auch in eine andere Richtig zu schauen... weil ich halt schon ne Weile was suche aber noch passendes gefunden habe... Manchmal hat ja jemand 'ne Idee...

**Reply by bynight66** [reply]
> Du kannst doch auch eigene Hardware und Software entwickeln und der Öffentlichkeit zur Verfügung stellen.
 
Kann ich leider nicht, obwohl mich Programmierung immer gereizt hat. Aber ich habe so viel andere Projekte, die mehr Zeit brauchen als ich leben werde, daß für so spezielle Programmierungen leider keinerlei Zeit bleibt.

> Oder nimm doch die herstellereigene DTU Pro.

Die hab ich doch seit Jahren. Deshalb suche ich ja nach Lösungen zur Kommunikation via Modbus... MQTT... Es gibt da sogar Ansätze/Lösungen die die Cloud abschaltet und die Daten der DTU Pro verwendet... vllt interessant für Freunde der cloudfreien Lösungen... ;-)

**Reply by bynight66** [reply]
> Das ist doch aber nur die Hardware. Software ist doch OpenDTU bzw. AhoyDTU. Gibt es alternative Firmware die mehr kann?

Womit wir wieder bei der unbeantworteten Ausgangsfrage wären... Wieso sind nur 10 bzw. 16 Inverter möglich?

**Reply by Solarteur** [reply]
Eine Antwort auf die gleiche Frage von dir gab es doch hier https://github.com/tbnobody/OpenDTU/discussions/1217 schon.

**Reply by huste511** [reply]
Sorry, dann bin ich zumindest blind.
Auf meinem fusion board war 84e947e installiert. Das ist doch eine alte original openDTU.
Allen Links den ich bisher gefolgt bin sind nur fusion doku mit links auf open DTU.
Hast Du mal nen passenen Link zur aktuellen fusion FW?

**Reply by bynight66** [reply]
Das war vor fast einem Jahr. Hätte ja sein können daß sich inzwischen mal etwas weiterentwickelt hat... :-)
Ok, also open DTU bringt mich hier nicht so recht weiter... aber zumindest sind mehr als 10 Wechselrichter nun thematisiert. Vllt frag ich in einem Jahr nochmal...  ;-) 
BtW... wem die Kosten wichtig sind... zur reinen Leistungsmessung mit grafischer Auswertung und Ein-/Ausschaltung der Wechselrichter kann ich die Tapo P110 empfehlen. Da sind in den 10 Euro bei Amazon Netzteil und Gehäuse schon mit drin. Cloudbasiert. Einbindung in HA dank vorhandener Integration ein Kinderspiel.

**Reply by huste511** [reply]
@schneeer
Danke für die Arbeit alle Links hier einzustellen. Genau auf den Seiten bin ich die Tage auch durch.

Das Problem kam wohl aus einer Zweideutigkeit bzw. widersprüchen.
Oben war ja genannt das fusion mehr können müßte. Das hatte ich auf die Firmware impliziert.
Auf der Webseite selber steht dann wieder "Bis zu 10 Wechselrichter gleichzeitig".
Zusätzlich war ich mit der Community Edition verwirrt -> "Beim Kauf dieses Produktes wird keinerlei technischer Support durch den Verkäufer geleistet" - So als gäbe es noch andere Software varianten.

Grund das es das Fusion *mit angepasster Firmware mehr könnte* ist der ESP32-S3-WROOM-1U?
Die 4MB SRAM sind das Entscheidende? An anderer Stelle las ich mal cie CPU sei zu schwach.
Verstand ich auch nicht so recht, da Boards meist 280Mhz haben und der WROOM auch als Singecore zu haben ist.

Vielleicht könnte man die WR-Beschränkung irgendwann man nicht hard coden sondern anhand des Boards variieren.
Ein kleiner Hinweis auf die Grenzen/Hardware in der Doku und ein Zähler/Limit und WR-Einrichtung wäre auch schön.

**Comment by Stefan-Matrei** [comment]
Hallo!
Ich habe 12 Wechselrichter, kann aber die Firmware nicht wie oben beschrieben auf 14 umstellen. Meine Kenntnisse sind dafür nicht ausreichend genug. Hat das jemand schon gemacht und könnte ich diese Firmware vielleicht auch mal testen.
Vielen Dank und schöne Grüße
